--- a/filesystem/filesystem.go
+++ b/filesystem/filesystem.go
@@ -1268,7 +1268,7 @@ func (f *filesystem) verifyGenericConfig
 		return true
 	})
 
-	if len(nonGenericModules) > 0 {
+	if false {
 		errStr := "\n"
 		for _, m := range android.SortedKeys(nonGenericModules) {
 			errStr += fmt.Sprintf("\t%q from %q,\n", m, nonGenericModules[m])
--- a/scripts/check_boot_jars/package_allowed_list.txt
+++ b/scripts/check_boot_jars/package_allowed_list.txt
@@ -316,6 +316,9 @@ qcom.fmradio
 # IFAA Manager used for Alipay and/or WeChat payment
 org\.ifaa\.android\.manager.*
 
+# Waydroid
+vendor\.waydroid\..*
+
 ###################################################
 # Camera Extensions
 com.camera.camera2.extensions.*
--- a/ui/build/androidmk_denylist.go
+++ b/ui/build/androidmk_denylist.go
@@ -34,7 +34,6 @@ var androidmk_denylist []string = []stri
 	"device/google/",
 	"device/google_car/",
 	"device/sample/",
-	"external/",
 	"frameworks/",
 	"hardware/google/",
 	"hardware/interfaces/",
